project/apps/web/hooks/useMe.ts

```diff
@@ -1,38 +1,40 @@
 "use client";
 
 import { useRouter } from "next/navigation";
-import { useCallback } from "react";
+import { useCallback, useEffect } from "react";
 
-import { useLoginState } from "@/contexts/LoginStateContext";
-import { useMutation, useSuspenseQuery } from "@tanstack/react-query";
-import { me, signOut } from "@/lib/api/auth";
-import { QUERY_KEYS } from "@/constants/queryKeys";
+import { useMutation } from "@tanstack/react-query";
+import { useAuthStore } from "@/lib/api/auth";
 
 export const useMe = () => {
   const router = useRouter();
-  const { data, error } = useSuspenseQuery({
-    queryKey: [QUERY_KEYS.Me],
-    queryFn: me,
+  const user = useAuthStore((state: any) => state.user);
+  const fetchUser = useAuthStore((state) => state.fetchUser);
+  const signOut = useAuthStore((state) => state.signOut);
```

Comment:
maybe implementing selectors can allow us to bypass needing the any? and also seems like the selectors allow us to access terms easier. [reference](https://zustand.docs.pmnd.rs/guides/auto-generating-selectors)